aioxmpp/structs.py

```diff
@@ -633,20 +634,11 @@ class JID(collections.namedtuple("JID", ["localpart", "domain", "resource"])):
 
     def __new__(cls, localpart, domain, resource, *, strict=True):
         if localpart:
-            localpart = nodeprep(
-                localpart,
-                allow_unassigned=not strict
-            )
+            localpart = localpart.encode('UsernameCaseMapped')
```

Comment:
No particular reason, I was just using this as an excuse to play around with the new PRECIS library a bit and didn't really think through the changes.